### PR TITLE
Do not fall back to ignoring empty directories for skip when empty

### PR DESCRIPTION
**Pull request by wolfs**
Removes the deprecation introduced in #18783 for Gradle 8.0.

### CODE REVIEW

File: subprojects/execution/src/main/java/org/gradle/internal/execution/fingerprint/FileCollectionFingerprinter.java [hunk 1]
```diff
@@ -38,7 +37,7 @@
     /**
      * Creates a fingerprint from the snapshot of a file collection.
      */
-    CurrentFileCollectionFingerprint fingerprint(FileSystemSnapshot snapshot, @Nullable FileCollectionFingerprint previousFingerprint);
+    CurrentFileCollectionFingerprint fingerprint(FileCollection files, @Nullable FileCollectionFingerprint previousFingerprint);
```
Comment (lptr) [inline subprojects/execution/src/main/java/org/gradle/internal/execution/fingerprint/FileCollectionFingerprinter.java:40]: Feels like we are taking a step back here. We should instead move towards requiring a snapshot all the time, no?

Comment (wolfs) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/fingerprint/FileCollectionFingerprinter.java:40]: That is what we could already do now on `master`, there is no reason to wait for 8.0.

So for me, the plan would be to be able to register a fingerprinting strategy, and then obtain that strategy via a spec. So `FileCollectionFingerprinter` would go away. Should we try to pursue this?

Comment (lptr) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/fingerprint/FileCollectionFingerprinter.java:40]: Let's do that!

**Comment by wolfs** [comment]
This is the PR for asciidoctor: https://github.com/asciidoctor/asciidoctor-gradle-plugin/pull/646

**Comment by wolfs** [comment]
Replaced by https://github.com/gradle/gradle/pull/21625.

**Pull request closed without merging**